### PR TITLE
Fix some bugs related to delayed triggers

### LINKED ISSUE #1229: Can't counter triggers from Effects
Example: trying to _Stifle_ (opponents) Epic trigger

Using `ValidTgts$ Card,Effect` seems to fix it.

@Hanmac 

your opinion?

### PR DESCRIPTION
**Pull request by tool4ever**
Closes #1229

@Hanmac 

I couldn't find anything that breaks it yet, but maybe you will :P

### CODE REVIEW

File: forge-game/src/main/java/forge/game/trigger/TriggerHandler.java [hunk 1]
```diff
@@ -541,8 +541,9 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
                 sa.changeText();
             }
         } else {
+            Player controller = delayedTriggers.contains(regtrig) ? sa.getActivatingPlayer() : host.getController();
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:544]: i don't like the `delayedTriggers.contains(regtrig)` check

could we maybe look for SpawningAbility ?

**Comment by tool4ever** [comment]
Fixed this case:
Exiling a creature with _Kaya, Ghost Assassin_ , then _Donate_ her to opponent
The return would have triggered on his upkeep

**Comment by Hanmac** [comment]
when touching delayed triggers, there is a rule that says that delayed trigger should not be affected by trigger additional times

**Comment by tool4ever** [comment]
yea, I already tried to break that but it seems `StaticAbilityPanharmonicon.handlePanharmonicon()` never gets called for delayed triggers but even then there's also a guard inside so should be safe...

File: forge-gui/res/cardsfolder/n/nimble_obstructionist.txt [hunk 2]
```diff
@@ -6,5 +6,5 @@ K:Flash
 K:Flying
 K:Cycling:2 U
 T:Mode$ Cycled | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When you cycle CARDNAME, counter target activated or triggered ability you don't control.
-SVar:TrigCounter:DB$ Counter | TargetType$ Activated,Triggered | ValidTgts$ Card.YouDontCtrl
+SVar:TrigCounter:DB$ Counter | TargetType$ Activated.YouDontCtrl,Triggered.YouDontCtrl | ValidTgts$ Card,Emblem
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/n/nimble_obstructionist.txt:9]: this probably needs a TgtPrompt

File: forge-gui/res/cardsfolder/s/stifle.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Name:Stifle
 ManaCost:U
 Types:Instant
-A:SP$ Counter | Cost$ U | TgtPrompt$ Select target Activated or Triggered Ability | ValidTgts$ Card | TargetType$ Activated,Triggered | SpellDescription$ Counter target activated or triggered ability.
+A:SP$ Counter | Cost$ U | TgtPrompt$ Select target Activated or Triggered Ability | ValidTgts$ Card,Emblem | TargetType$ Activated,Triggered | SpellDescription$ Counter target activated or triggered ability.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/s/stifle.txt:4]: `Activated or Triggered` -> `activated or triggered`

File: forge-gui/res/cardsfolder/s/strionic_resonator.txt [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Name:Strionic Resonator
 ManaCost:2
 Types:Artifact
-A:AB$ CopySpellAbility | Cost$ 2 T | TargetType$ Triggered.YouCtrl | ValidTgts$ Card | MayChooseTarget$ True | SpellDescription$ Copy target triggered ability you control. You may choose new targets for the copy.
+A:AB$ CopySpellAbility | Cost$ 2 T | TargetType$ Triggered.YouCtrl | ValidTgts$ Card,Emblem | MayChooseTarget$ True | SpellDescription$ Copy target triggered ability you control. You may choose new targets for the copy.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/s/strionic_resonator.txt:4]: TgtPrompt

File: forge-gui/res/cardsfolder/t/trickbind.txt [hunk 5]
```diff
@@ -2,7 +2,7 @@ Name:Trickbind
 ManaCost:1 U
 Types:Instant
 K:Split second
-A:SP$ Counter | Cost$ 1 U | TargetType$ Activated,Triggered | TgtPrompt$ Select target Activated or Triggered Ability | RememberCountered$ True | ValidTgts$ Card | SubAbility$ DBEffect | SpellDescription$ Counter target activated or triggered ability. If a permanent's ability is countered this way, activated abilities of that permanent can't be activated this turn. (Mana abilities can't be targeted.)
+A:SP$ Counter | Cost$ 1 U | TargetType$ Activated,Triggered | TgtPrompt$ Select target Activated or Triggered Ability | RememberCountered$ True | ValidTgts$ Card,Emblem | SubAbility$ DBEffect | SpellDescription$ Counter target activated or triggered ability. If a permanent's ability is countered this way, activated abilities of that permanent can't be activated this turn. (Mana abilities can't be targeted.)
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/t/trickbind.txt:5]: Decapitalize again

**Pull request merged**